### PR TITLE
pbzip2 embeds g++ for compilation in Makefile

### PR DESCRIPTION
**Pull request by mitchty**
On osx, replace g++ with clang++ so it is able to build.

### CODE REVIEW

File: pkgs/tools/compression/pbzip2/default.nix [hunk 1]
```diff
@@ -11,9 +11,15 @@ stdenv.mkDerivation rec {
     sha256 = "1i7rql77ac33lz7lzrjyl9b16mqizmdkb8hv295a493f7vh1nhmx";
   };
 
-  buildInputs = [ bzip2 ];
+  buildInputs = [ bzip2 ] ++ 
+    stdenv.lib.optional stdenv.isDarwin [ clang ];
+
+  preBuild = stdenv.lib.optionalString stdenv.isDarwin ''
+    substituteInPlace Makefile --replace g++ clang++
```
Comment (nbp) [inline pkgs/tools/compression/pbzip2/default.nix]: Shouldn't we replace it by `c++` when stdenv is not based on gcc?

Comment (mitchty) [inline pkgs/tools/compression/pbzip2/default.nix]: Couldn't find anything in the stdenv setup to indicate how to determine stdenv isn't built on gcc. Just switched it to replace to c++ and things seem to still work at least. So no need for this to be clang/osx specific.

Comment (spwhitt) [inline pkgs/tools/compression/pbzip2/default.nix]: You can use: `!stdenv.cc.isGNU`

**Comment by vcunat** [comment]
Well, our gcc does contain the `c++` symlink, so that name should work everywhere. I pushed that for simplicity.

**Pull request closed without merging**